Assets/Python/RFCUtils.py

```diff
@@ -53,39 +53,16 @@ def getRandomCity(self, iPlayer):
 
 	# Leoreth - finds an adjacent land plot without enemy units that's closest to the player's capital (for the Roman UP)
 	def findNearestLandPlot(self, tPlot, iPlayer):
-		x, y = tPlot
 		plotList = []
 
-		for i in range(x - 1, x + 2):
-			for j in range(y - 1, y + 2):
-				pPlot = gc.getMap().plot( i, j )
-				if (not pPlot.isWater() and not pPlot.isPeak()):
-					if ( not pPlot.isUnit() ):
-						plotList.append(pPlot)
-
-		#if (len(plotList) > 0):
-		#	iDistance = 1000
-		#	pCapital = gc.getPlayer(iPlayer).getCapitalCity()
-		#	iCapX = pCapital.getX()
-		#	iCapY = pCapital.getY()
-		#	for plot in plotList:
-		#		plotX = plot.getX()
-		#		plotY = plot.getY()
-		#		iTempDist = self.calculateDistance(iCapX, plotX, iCapY, plotY)
-#
-		#		if iTempDist < iDistance:
-		#			nearestPlot = plot
-		#			iDistance = iTempDist
-
-		#	return nearestPlot
-
-		#return (None)
-
-		if (len(plotList) > 0):
-			rndNum = gc.getGame().getSorenRandNum(len(plotList), 'land plot')
-			result = plotList[rndNum]
-			if (result):
-				return ((result.getX(), result.getY()))
+		for (x, y) in self.surroundingPlots(tPlot):
+			pPlot = gc.getMap().plot(x, y)
+			if not pPlot.isWater() and not pPlot.isPeak():
+				if not pPlot.isUnit():
+					plotList.append(pPlot)
+
+		if plotList:
+			return self.getRandomEntry(plotList)

```

Comment:
This returns a PyPlot object, not the corresponding (x, y) tuple as intended. I suggest putting the tuple into the list when it is created above right away, it never was a good idea to fill it with objects.